appscale/tools/appscale.py

```diff
@@ -337,7 +337,11 @@ def valid_ssh_key(self, config, run_instances_opts):
     if not os.path.exists(ssh_key_location):
       return False
 
-    all_ips = LocalState.get_all_public_ips(keyname)
+    try:
+      all_ips = LocalState.get_all_public_ips(keyname)
+    except BadConfigurationException:
+      # If this is an upgrade from 3.1.0, there may not be a locations JSON.
+      all_ips = set(run_instances_opts.ips.values())
```

Comment:
This function is only run when 'infrastructure' is not in the AppScalefile. I'm open to suggestions for making it more defensive, though.